crates/turborepo-scm/src/package_deps.rs

```diff
@@ -99,11 +116,52 @@ impl Git {
         turbo_root: &AbsoluteSystemPath,
         package_path: &AnchoredSystemPath,
         inputs: &[S],
+        include_default_files: bool,
     ) -> Result<GitHashes, Error> {
+        if include_default_files && !inputs.is_empty() {
+            // collect the default files and the inputs
+            let default_file_hashes =
+                self.get_package_file_hashes_from_index(turbo_root, package_path)?;
+
+            // we need to get hashes for excludes separately so we can remove them from the
+            // defaults later on
+            let mut includes = Vec::new();
+            let mut excludes = Vec::new();
+            for input in inputs {
+                let input_str = input.as_ref();
+                if let Some(exclude) = input_str.strip_prefix('!') {
+                    excludes.push(exclude);
+                } else {
+                    includes.push(input_str);
+                }
+            }
```

Comment:
don't we have this logic somewhere already? We don't do this up front in turbo.json parsing?